plugin/associations.go

```diff
@@ -201,6 +222,66 @@ func (p *OrmPlugin) parseBelongsTo(msg *generator.Descriptor, child *OrmableType
 	}
 }
 
+func (p *OrmPlugin) parseManyToMany(msg *generator.Descriptor, ormable *OrmableType, fieldName string, fieldType string, assoc *OrmableType, opts *gorm.GormFieldOptions) {
+	typeName := generator.CamelCaseSlice(msg.TypeName())
+	mtm := opts.GetManyToMany()
+	if mtm == nil {
+		mtm = &gorm.ManyToManyOptions{}
+		opts.Association = &gorm.GormFieldOptions_ManyToMany{mtm}
+	}
+
+	var foreignKey *Field
+	var foreignKeyName string
+	if foreignKeyName = generator.CamelCase(mtm.GetForeignkey()); foreignKeyName == "" {
+		foreignKeyName, foreignKey = p.findPrimaryKey(ormable)
+	} else {
+		var ok bool
+		foreignKey, ok = ormable.Fields[foreignKeyName]
+		if !ok {
+			p.Fail("Missing", foreignKeyName, "field in", ormable.Name, ".")
+		}
+	}
+	mtm.Foreignkey = &foreignKeyName
+	var assocKey *Field
+	var assocKeyName string
+	if assocKeyName = generator.CamelCase(mtm.GetAssociationForeignkey()); assocKeyName == "" {
+		assocKeyName, assocKey = p.findPrimaryKey(assoc)
+	} else {
+		var ok bool
+		assocKey, ok = assoc.Fields[assocKeyName]
+		if !ok {
+			p.Fail("Missing", assocKeyName, "field in", assoc.Name, ".")
+		}
+	}
+	mtm.AssociationForeignkey = &assocKeyName
+	if foreignKey.Type != assocKey.Type {
+		p.Fail("Type mismatch of foreignkey and association foreignkey in", ormable.Name, fieldName, "association.")
+	}
+	var jt string
+	if jt = generator.CamelCase(mtm.GetJointable()); jt == "" {
+		if p.countManyToManyAssociationDimension(msg, fieldType) == 1 && typeName != fieldType {
+			jt = jgorm.ToDBName(typeName + fieldType)
+		} else {
+			jt = jgorm.ToDBName(typeName + inflection.Singular(fieldName))
```

Comment:
Is the `Singular` here more intuitive? That results in `user_language` instead of `user_languages`, right? The gorm example http://gorm.io/docs/many_to_many.html has the plural instead. I'm open to either, but was wondering about the reasoning.

Reply:
`user_language` appeared more natural for me on first sight, but it seems that everybody [suggests](https://stackoverflow.com/questions/1764483/sql-join-table-naming-convention) `user_languages` instead. Let's do it this way.